### PR TITLE
ledger: signing support

### PR DESCRIPTION
**Pull request by 0xwml**

### CODE REVIEW

File: src/ledger/ledger.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  // @ts-ignore
+  const transport = await TransportNodeHid.default.open('');
+
+  // Initialize the AptosLedgerClient with the transport
+  const ledgerClient = new AptosLedgerClient(transport);
+
+  return ledgerClient;
+}
+
+export async function initLedgerSigner(ledgerIndex: number = 0): Promise<LedgerSigner> {
```
Comment (0xbe1) [inline src/ledger/ledger.ts:20]: what does ledgerIndex mean?

File: src/signing.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { loadAccount } from './accounts.js';
+import { closeLedger, initLedgerSigner, signerToAddress } from './ledger/ledger.js';
+import { Account, AnyRawTransaction, Aptos } from '@aptos-labs/ts-sdk';
+import LedgerSigner from './ledger/LedgerSigner';
+
+// Pending Txns
+
+export async function getSender(
+  options: any
```
Comment (0xbe1) [inline src/signing.ts:9]: we should try to avoid `any`

File: src/ledger/ledger.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  return signer;
+}
+
+export function signerToAddress(signer: LedgerSigner) {
+  // @ts-ignore
+  return publicKeyToAddress(signer.publicKey.bcsToBytes().slice(1)).toString();
+}
+
+export function getHDPath(ledgerIndex: number) {
+  return `m/44'/637'/${ledgerIndex}'/0'/0'`;
```
Comment (0xbe1) [inline src/ledger/ledger.ts:40]: what does this mean?

**Review by 0xbe1 — COMMENTED** [review]
can you test and ensure ledger work for all these operations, and the code change doesn't break operations using profile?

File: src/ledger/ledger.ts (hunk 1 above)
Comment (0xwml) [reply src/ledger/ledger.ts:20]: your ledger can have multiple addresses, which are all derived from a HD path (HD = Hierarchically Deterministic). `Ledger index` lets the user specify which account index they intend on using for the transaction

File: src/ledger/ledger.ts (hunk 3 above)
Comment (0xwml) [reply src/ledger/ledger.ts:40]: added a bunch of docs

**Comment by 0xwml** [comment]
> can you test and ensure ledger work for all these operations, and the code change doesn't break operations using profile?

I have verified that 

```
* Execute
* Owners add
* Owners remove
* Owners swap
* Vote
* Propose
```

are all working as expected

**Review by 0xbe1 — APPROVED** [review]
lgtm

**Pull request merged**